### ISSUE TITLE
Error from Etcd on full disk (crash and burn-mode)

### ISSUE THREAD

**Issue by haf** [deis/deis#3235]
My testing continues with another in-production frequent occurrence; a full disk.

First, start a vagrant cluster.

Fill the disk of the 'master' deis-01.

Force-restart all machines (it could happen!), breaking any/all quorums.

When all nodes are back up and responding to ssh, run `deisctl start platform`.

Nice errors messages:

```
➜  deis-vagrant git:(docs/cadvisor) ✗ deisctl start platform
● ▴ ■
■ ● ▴ Starting Deis...
▴ ■ ●

Storage subsystem...
INFO client.go:297: Response 500 Internal Server Error from http://127.0.0.1:4001/ unusable
```

From inside a machine that hasn't got full disk (deis-02):

```
Mar 06 15:02:47 localhost kernel: random: nonblocking pool is initialized
Mar 06 15:02:47 localhost systemd-journal[401]: Journal started
...skipping...
# snip, below line repeated hundreds of times
Mar 08 16:29:07 deis-02 etcd[567]: [etcd] Mar  8 16:29:07.955 WARNING   | transporter.ss.decoding.error:proto: field/encoding mismatch: wrong type for field
```

From deis-01:

```
$ journalctl -fu etcd
-- Logs begin at Fri 2015-03-06 15:02:26 UTC. --
Mar 08 16:32:04 deis-01 etcd[568]: [etcd] Mar  8 16:32:04.769 WARNING   | [ss] Error: nil response
Mar 08 16:32:05 deis-01 etcd[568]: [etcd] Mar  8 16:32:05.269 WARNING   | [ss] Error: nil response
```

From deis-03:

```
$ journalctl -fu etcd
-- Logs begin at Fri 2015-03-06 15:03:08 UTC. --
Mar 08 16:25:57 deis-03 etcd[567]: [etcd] Mar  8 16:25:57.661 INFO      | peer server [name 169914f25a9b47f38a8f4da8eee9a714, listen on :7001, advertised url http://172.17.8.102:7001]
Mar 08 16:25:57 deis-03 etcd[567]: [etcd] Mar  8 16:25:57.661 INFO      | 169914f25a9b47f38a8f4da8eee9a714 starting in peer mode
Mar 08 16:25:57 deis-03 etcd[567]: [etcd] Mar  8 16:25:57.661 INFO      | 169914f25a9b47f38a8f4da8eee9a714: state changed from 'initialized' to 'follower'.
Mar 08 16:26:00 deis-03 etcd[567]: [etcd] Mar  8 16:26:00.179 INFO      | 169914f25a9b47f38a8f4da8eee9a714: state changed from 'follower' to 'candidate'.
Mar 08 16:26:00 deis-03 etcd[567]: [etcd] Mar  8 16:26:00.181 INFO      | removed during cluster re-configuration
Mar 08 16:26:00 deis-03 etcd[567]: [etcd] Mar  8 16:26:00.181 INFO      | 169914f25a9b47f38a8f4da8eee9a714: state changed from 'candidate' to 'follower'.
Mar 08 16:26:00 deis-03 etcd[567]: [etcd] Mar  8 16:26:00.181 INFO      | 169914f25a9b47f38a8f4da8eee9a714: term #18446744073709551615 started.
Mar 08 16:26:00 deis-03 etcd[567]: [etcd] Mar  8 16:26:00.181 INFO      | 169914f25a9b47f38a8f4da8eee9a714: state changed from 'follower' to 'stopped'.
Mar 08 16:26:00 deis-03 etcd[567]: [etcd] Mar  8 16:26:00.182 WARNING   | transporter.vr.decoding.error:proto: field/encoding mismatch: wrong type for field
Mar 08 16:26:00 deis-03 etcd[567]: [etcd] Mar  8 16:26:00.182 INFO      | 169914f25a9b47f38a8f4da8eee9a714: state changed from 'stopped' to 'stopped'.
```

**Comment by haf** [comment]
Trying to rescue the situation by destroying the node with full disk. Then `vagrant up deis-01`:

```
➜  deis-vagrant git:(docs/cadvisor) ✗ vagrant up deis-01
Bringing machine 'deis-01' up with 'virtualbox' provider...
==> deis-01: Running triggers before up...
==> deis-01: Checking if box 'coreos-stable' is up to date...
==> deis-01: VirtualBox VM is already running.
➜  deis-vagrant git:(docs/cadvisor) ✗ vagrant ssh deis-01
 * *    *   *****    ddddd   eeeeeee iiiiiii   ssss
*   *  * *  *   *     d   d   e    e    i     s    s
 * *  ***** *****     d    d  e         i    s
*****  * *    *       d     d e         i     s
*   * *   *  * *      d     d eee       i      sss
*****  * *  *****     d     d e         i         s
  *   *****  * *      d    d  e         i          s
 * *  *   * *   *     d   d   e    e    i    s    s
***** *****  * *     ddddd   eeeeeee iiiiiii  ssss

Welcome to Deis			Powered by CoreOS
core@deis-01 ~ $ docker ps
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
core@deis-01 ~ $ deis-debug-logs
--- VERSIONS ---
CoreOS 557.2.0
Deis v1.4.1
etcd version 0.4.6
fleetd version 0.9.0

--- SYSTEM STATUS ---
-- Logs begin at Sun 2015-03-08 16:35:16 UTC, end at Sun 2015-03-08 16:36:05 UTC. --
Mar 08 16:35:30 deis-01 systemd[1]: Starting etcd...
Mar 08 16:35:30 deis-01 systemd[1]: Started etcd.
Mar 08 16:35:30 deis-01 etcd[980]: [etcd] Mar  8 16:35:30.215 INFO      | The path /var/lib/etcd/log is in btrfs
Mar 08 16:35:30 deis-01 etcd[980]: [etcd] Mar  8 16:35:30.215 INFO      | Set NOCOW to path /var/lib/etcd/log succeeded
Mar 08 16:35:30 deis-01 etcd[980]: [etcd] Mar  8 16:35:30.216 INFO      | Discovery via https://discovery.etcd.io using prefix /1437e98a40dc96c58603e45f02cf3b52.
Mar 08 16:35:31 deis-01 etcd[980]: [etcd] Mar  8 16:35:31.916 INFO      | Discovery found peers [http://172.17.8.100:7001 http://172.17.8.101:7001 http://172.17.8.102:7001]
Mar 08 16:35:31 deis-01 etcd[980]: [etcd] Mar  8 16:35:31.916 INFO      | Discovery fetched back peer list: [172.17.8.100:7001 172.17.8.101:7001 172.17.8.102:7001]
Mar 08 16:35:31 deis-01 etcd[980]: [etcd] Mar  8 16:35:31.917 INFO      | 6cf379e0f02d4f30a9ab7bf2e8600504 attempted to join via 172.17.8.100:7001 failed: fail checking join version: Client Internal Error (Get http://172.17.8.100:7001/version: dial tcp 172.17.8.100:7001: connection refused)
Mar 08 16:35:31 deis-01 etcd[980]: [etcd] Mar  8 16:35:31.920 INFO      | Send Join Request to http://172.17.8.101:7001/join
Mar 08 16:35:36 deis-01 etcd[980]: [etcd] Mar  8 16:35:36.421 INFO      | 6cf379e0f02d4f30a9ab7bf2e8600504 attempted to join via 172.17.8.101:7001 failed: fail on join request: Client Internal Error (Put http://172.17.8.101:7001/join: net/http: timeout awaiting response headers)
Mar 08 16:35:36 deis-01 etcd[980]: [etcd] Mar  8 16:35:36.425 INFO      | Send Join Request to http://172.17.8.102:7001/join
Mar 08 16:35:36 deis-01 etcd[980]: [etcd] Mar  8 16:35:36.426 INFO      | 6cf379e0f02d4f30a9ab7bf2e8600504 attempted to join via 172.17.8.102:7001 failed: fail on join request: Raft Internal Error ()
Mar 08 16:35:36 deis-01 etcd[980]: [etcd] Mar  8 16:35:36.426 INFO      | 6cf379e0f02d4f30a9ab7bf2e8600504 is unable to join the cluster using any of the peers [172.17.8.100:7001 172.17.8.101:7001 172.17.8.102:7001] at 0th time. Retrying in 3.0 seconds
Mar 08 16:35:39 deis-01 etcd[980]: [etcd] Mar  8 16:35:39.428 INFO      | 6cf379e0f02d4f30a9ab7bf2e8600504 attempted to join via 172.17.8.100:7001 failed: fail checking join version: Client Internal Error (Get http://172.17.8.100:7001/version: dial tcp 172.17.8.100:7001: connection refused)
Mar 08 16:35:39 deis-01 etcd[980]: [etcd] Mar  8 16:35:39.431 INFO      | Send Join Request to http://172.17.8.101:7001/join
Mar 08 16:35:43 deis-01 etcd[980]: [etcd] Mar  8 16:35:43.932 INFO      | 6cf379e0f02d4f30a9ab7bf2e8600504 attempted to join via 172.17.8.101:7001 failed: fail on join request: Client Internal Error (Put http://172.17.8.101:7001/join: net/http: timeout awaiting response headers)
Mar 08 16:35:43 deis-01 etcd[980]: [etcd] Mar  8 16:35:43.937 INFO      | Send Join Request to http://172.17.8.102:7001/join
Mar 08 16:35:43 deis-01 etcd[980]: [etcd] Mar  8 16:35:43.938 INFO      | 6cf379e0f02d4f30a9ab7bf2e8600504 attempted to join via 172.17.8.102:7001 failed: fail on join request: Raft Internal Error ()
Mar 08 16:35:43 deis-01 etcd[980]: [etcd] Mar  8 16:35:43.939 INFO      | 6cf379e0f02d4f30a9ab7bf2e8600504 is unable to join the cluster using any of the peers [172.17.8.100:7001 172.17.8.101:7001 172.17.8.102:7001] at 1th time. Retrying in 3.0 seconds
Mar 08 16:35:46 deis-01 etcd[980]: [etcd] Mar  8 16:35:46.940 INFO      | 6cf379e0f02d4f30a9ab7bf2e8600504 attempted to join via 172.17.8.100:7001 failed: fail checking join version: Client Internal Error (Get http://172.17.8.100:7001/version: dial tcp 172.17.8.100:7001: connection refused)
Mar 08 16:35:46 deis-01 etcd[980]: [etcd] Mar  8 16:35:46.945 INFO      | Send Join Request to http://172.17.8.101:7001/join
Mar 08 16:35:51 deis-01 etcd[980]: [etcd] Mar  8 16:35:51.446 INFO      | 6cf379e0f02d4f30a9ab7bf2e8600504 attempted to join via 172.17.8.101:7001 failed: fail on join request: Client Internal Error (Put http://172.17.8.101:7001/join: net/http: timeout awaiting response headers)
Mar 08 16:35:51 deis-01 etcd[980]: [etcd] Mar  8 16:35:51.449 INFO      | Send Join Request to http://172.17.8.102:7001/join
Mar 08 16:35:51 deis-01 systemd[1]: etcd.service: main process exited, code=exited, status=1/FAILURE
Mar 08 16:35:51 deis-01 systemd[1]: Unit etcd.service entered failed state.
Mar 08 16:35:51 deis-01 systemd[1]: etcd.service failed.
Mar 08 16:35:51 deis-01 etcd[980]: [etcd] Mar  8 16:35:51.449 INFO      | 6cf379e0f02d4f30a9ab7bf2e8600504 attempted to join via 172.17.8.102:7001 failed: fail on join request: Raft Internal Error ()
Mar 08 16:35:51 deis-01 etcd[980]: [etcd] Mar  8 16:35:51.449 WARNING   | 6cf379e0f02d4f30a9ab7bf2e8600504 cannot connect to existing peers [172.17.8.100:7001 172.17.8.101:7001 172.17.8.102:7001]: fail joining the cluster via given peers after 3 retries
Mar 08 16:35:51 deis-01 etcd[980]: [etcd] Mar  8 16:35:51.449 CRITICAL  | fail joining the cluster via given peers after 3 retries
Mar 08 16:36:01 deis-01 systemd[1]: etcd.service holdoff time over, scheduling restart.
Mar 08 16:36:01 deis-01 systemd[1]: Stopping etcd...
Mar 08 16:36:01 deis-01 systemd[1]: Starting etcd...
Mar 08 16:36:01 deis-01 systemd[1]: Started etcd.
Mar 08 16:36:01 deis-01 etcd[1044]: [etcd] Mar  8 16:36:01.565 INFO      | The path /var/lib/etcd/log is in btrfs
Mar 08 16:36:01 deis-01 etcd[1044]: [etcd] Mar  8 16:36:01.565 INFO      | Set NOCOW to path /var/lib/etcd/log succeeded
Mar 08 16:36:01 deis-01 etcd[1044]: [etcd] Mar  8 16:36:01.565 INFO      | Discovery via https://discovery.etcd.io using prefix /1437e98a40dc96c58603e45f02cf3b52.
Mar 08 16:36:05 deis-01 etcd[1044]: [etcd] Mar  8 16:36:05.260 WARNING   | Discovery encountered an error: 501: All the given peers are not reachable (Tried to connect to each peer twice and failed) [0]
Mar 08 16:36:05 deis-01 etcd[1044]: [etcd] Mar  8 16:36:05.260 WARNING   | 6cf379e0f02d4f30a9ab7bf2e8600504 failed to connect discovery service[https://discovery.etcd.io/1437e98a40dc96c58603e45f02cf3b52]: 501: All the given peers are not reachable (Tried to connect to each peer twice and failed) [0]
Mar 08 16:36:05 deis-01 etcd[1044]: [etcd] Mar  8 16:36:05.260 CRITICAL  | 6cf379e0f02d4f30a9ab7bf2e8600504, the new instance, must register itself to discovery service as required
Mar 08 16:36:05 deis-01 systemd[1]: etcd.service: main process exited, code=exited, status=1/FAILURE
Mar 08 16:36:05 deis-01 systemd[1]: Unit etcd.service entered failed state.
Mar 08 16:36:05 deis-01 systemd[1]: etcd.service failed.
-- Logs begin at Sun 2015-03-08 16:35:16 UTC, end at Sun 2015-03-08 16:36:05 UTC. --
Mar 08 16:35:42 deis-01 fleetd[981]: INFO client.go:291: Failed getting response from http://localhost:4001/: dial tcp 127.0.0.1:4001: connection refused
Mar 08 16:35:42 deis-01 fleetd[981]: ERROR client.go:213: Unable to get result for {Create /_coreos.com/fleet/machines/6cf379e0f02d4f30a9ab7bf2e8600504/object}, retrying in 1s
Mar 08 16:35:45 deis-01 fleetd[981]: INFO client.go:291: Failed getting response from http://localhost:4001/: dial tcp 127.0.0.1:4001: connection refused
Mar 08 16:35:45 deis-01 fleetd[981]: ERROR client.go:213: Unable to get result for {Update /_coreos.com/fleet/machines/6cf379e0f02d4f30a9ab7bf2e8600504/object}, retrying in 100ms
Mar 08 16:35:45 deis-01 fleetd[981]: INFO client.go:291: Failed getting response from http://localhost:4001/: dial tcp 127.0.0.1:4001: connection refused
Mar 08 16:35:45 deis-01 fleetd[981]: ERROR client.go:213: Unable to get result for {Update /_coreos.com/fleet/machines/6cf379e0f02d4f30a9ab7bf2e8600504/object}, retrying in 200ms
Mar 08 16:35:45 deis-01 fleetd[981]: INFO client.go:291: Failed getting response from http://localhost:4001/: dial tcp 127.0.0.1:4001: connection refused
Mar 08 16:35:45 deis-01 fleetd[981]: ERROR client.go:213: Unable to get result for {Update /_coreos.com/fleet/machines/6cf379e0f02d4f30a9ab7bf2e8600504/object}, retrying in 400ms
Mar 08 16:35:45 deis-01 fleetd[981]: INFO client.go:291: Failed getting response from http://localhost:4001/: dial tcp 127.0.0.1:4001: connection refused
Mar 08 16:35:45 deis-01 fleetd[981]: ERROR client.go:213: Unable to get result for {Update /_coreos.com/fleet/machines/6cf379e0f02d4f30a9ab7bf2e8600504/object}, retrying in 800ms
Mar 08 16:35:46 deis-01 fleetd[981]: INFO client.go:291: Failed getting response from http://localhost:4001/: dial tcp 127.0.0.1:4001: connection refused
Mar 08 16:35:46 deis-01 fleetd[981]: ERROR client.go:213: Unable to get result for {Update /_coreos.com/fleet/machines/6cf379e0f02d4f30a9ab7bf2e8600504/object}, retrying in 1s
Mar 08 16:35:47 deis-01 fleetd[981]: INFO client.go:291: Failed getting response from http://localhost:4001/: dial tcp 127.0.0.1:4001: connection refused
Mar 08 16:35:47 deis-01 fleetd[981]: ERROR client.go:213: Unable to get result for {Update /_coreos.com/fleet/machines/6cf379e0f02d4f30a9ab7bf2e8600504/object}, retrying in 1s
Mar 08 16:35:48 deis-01 fleetd[981]: INFO client.go:291: Failed getting response from http://localhost:4001/: dial tcp 127.0.0.1:4001: connection refused
Mar 08 16:35:48 deis-01 fleetd[981]: ERROR client.go:213: Unable to get result for {Create /_coreos.com/fleet/machines/6cf379e0f02d4f30a9ab7bf2e8600504/object}, retrying in 100ms
Mar 08 16:35:48 deis-01 fleetd[981]: INFO client.go:291: Failed getting response from http://localhost:4001/: dial tcp 127.0.0.1:4001: connection refused
Mar 08 16:35:48 deis-01 fleetd[981]: ERROR client.go:213: Unable to get result for {Create /_coreos.com/fleet/machines/6cf379e0f02d4f30a9ab7bf2e8600504/object}, retrying in 200ms
Mar 08 16:35:48 deis-01 fleetd[981]: INFO client.go:291: Failed getting response from http://localhost:4001/: dial tcp 127.0.0.1:4001: connection refused
Mar 08 16:35:48 deis-01 fleetd[981]: ERROR client.go:213: Unable to get result for {Create /_coreos.com/fleet/machines/6cf379e0f02d4f30a9ab7bf2e8600504/object}, retrying in 400ms
Mar 08 16:35:48 deis-01 fleetd[981]: INFO client.go:291: Failed getting response from http://localhost:4001/: dial tcp 127.0.0.1:4001: connection refused
Mar 08 16:35:48 deis-01 fleetd[981]: ERROR client.go:213: Unable to get result for {Create /_coreos.com/fleet/machines/6cf379e0f02d4f30a9ab7bf2e8600504/object}, retrying in 800ms
Mar 08 16:35:49 deis-01 fleetd[981]: INFO client.go:291: Failed getting response from http://localhost:4001/: dial tcp 127.0.0.1:4001: connection refused
Mar 08 16:35:49 deis-01 fleetd[981]: ERROR client.go:213: Unable to get result for {Create /_coreos.com/fleet/machines/6cf379e0f02d4f30a9ab7bf2e8600504/object}, retrying in 1s
Mar 08 16:35:50 deis-01 fleetd[981]: INFO client.go:291: Failed getting response from http://localhost:4001/: dial tcp 127.0.0.1:4001: connection refused
Mar 08 16:35:50 deis-01 fleetd[981]: ERROR client.go:213: Unable to get result for {Create /_coreos.com/fleet/machines/6cf379e0f02d4f30a9ab7bf2e8600504/object}, retrying in 1s
Mar 08 16:35:55 deis-01 fleetd[981]: INFO client.go:291: Failed getting response from http://localhost:4001/: dial tcp 127.0.0.1:4001: connection refused
Mar 08 16:35:55 deis-01 fleetd[981]: ERROR client.go:213: Unable to get result for {Update /_coreos.com/fleet/machines/6cf379e0f02d4f30a9ab7bf2e8600504/object}, retrying in 100ms
Mar 08 16:35:55 deis-01 fleetd[981]: INFO client.go:291: Failed getting response from http://localhost:4001/: dial tcp 127.0.0.1:4001: connection refused
Mar 08 16:35:55 deis-01 fleetd[981]: ERROR client.go:213: Unable to get result for {Update /_coreos.com/fleet/machines/6cf379e0f02d4f30a9ab7bf2e8600504/object}, retrying in 200ms
Mar 08 16:35:55 deis-01 fleetd[981]: INFO client.go:291: Failed getting response from http://localhost:4001/: dial tcp 127.0.0.1:4001: connection refused
Mar 08 16:35:55 deis-01 fleetd[981]: ERROR client.go:213: Unable to get result for {Update /_coreos.com/fleet/machines/6cf379e0f02d4f30a9ab7bf2e8600504/object}, retrying in 400ms
Mar 08 16:35:55 deis-01 fleetd[981]: INFO client.go:291: Failed getting response from http://localhost:4001/: dial tcp 127.0.0.1:4001: connection refused
Mar 08 16:35:55 deis-01 fleetd[981]: ERROR client.go:213: Unable to get result for {Update /_coreos.com/fleet/machines/6cf379e0f02d4f30a9ab7bf2e8600504/object}, retrying in 800ms
Mar 08 16:35:56 deis-01 fleetd[981]: INFO client.go:291: Failed getting response from http://localhost:4001/: dial tcp 127.0.0.1:4001: connection refused
Mar 08 16:35:56 deis-01 fleetd[981]: ERROR client.go:213: Unable to get result for {Update /_coreos.com/fleet/machines/6cf379e0f02d4f30a9ab7bf2e8600504/object}, retrying in 1s
Mar 08 16:35:57 deis-01 fleetd[981]: INFO client.go:291: Failed getting response from http://localhost:4001/: dial tcp 127.0.0.1:4001: connection refused
Mar 08 16:35:57 deis-01 fleetd[981]: ERROR client.go:213: Unable to get result for {Update /_coreos.com/fleet/machines/6cf379e0f02d4f30a9ab7bf2e8600504/object}, retrying in 1s
Mar 08 16:35:58 deis-01 fleetd[981]: INFO client.go:291: Failed getting response from http://localhost:4001/: dial tcp 127.0.0.1:4001: connection refused
Mar 08 16:35:58 deis-01 fleetd[981]: ERROR client.go:213: Unable to get result for {Create /_coreos.com/fleet/machines/6cf379e0f02d4f30a9ab7bf2e8600504/object}, retrying in 100ms
Mar 08 16:35:58 deis-01 fleetd[981]: INFO client.go:291: Failed getting response from http://localhost:4001/: dial tcp 127.0.0.1:4001: connection refused
Mar 08 16:35:58 deis-01 fleetd[981]: ERROR client.go:213: Unable to get result for {Create /_coreos.com/fleet/machines/6cf379e0f02d4f30a9ab7bf2e8600504/object}, retrying in 200ms
Mar 08 16:35:58 deis-01 fleetd[981]: INFO client.go:291: Failed getting response from http://localhost:4001/: dial tcp 127.0.0.1:4001: connection refused
Mar 08 16:35:58 deis-01 fleetd[981]: ERROR client.go:213: Unable to get result for {Create /_coreos.com/fleet/machines/6cf379e0f02d4f30a9ab7bf2e8600504/object}, retrying in 400ms
Mar 08 16:35:58 deis-01 fleetd[981]: INFO client.go:291: Failed getting response from http://localhost:4001/: dial tcp 127.0.0.1:4001: connection refused
Mar 08 16:35:58 deis-01 fleetd[981]: ERROR client.go:213: Unable to get result for {Create /_coreos.com/fleet/machines/6cf379e0f02d4f30a9ab7bf2e8600504/object}, retrying in 800ms
Mar 08 16:35:59 deis-01 fleetd[981]: INFO client.go:291: Failed getting response from http://localhost:4001/: dial tcp 127.0.0.1:4001: connection refused
Mar 08 16:35:59 deis-01 fleetd[981]: ERROR client.go:213: Unable to get result for {Create /_coreos.com/fleet/machines/6cf379e0f02d4f30a9ab7bf2e8600504/object}, retrying in 1s
Mar 08 16:36:00 deis-01 fleetd[981]: INFO client.go:291: Failed getting response from http://localhost:4001/: dial tcp 127.0.0.1:4001: connection refused
Mar 08 16:36:00 deis-01 fleetd[981]: ERROR client.go:213: Unable to get result for {Create /_coreos.com/fleet/machines/6cf379e0f02d4f30a9ab7bf2e8600504/object}, retrying in 1s

--- DEIS STATUS ---
/opt/bin/deis-debug-logs: line 18: deisctl: command not found
Error:  Cannot sync with the cluster using peers 127.0.0.1:4001
```

And trying to re-install the platform on that node:

```
deisctl config platform set domain=local3.deisapp.com
Error: 501: All the given peers are not reachable (Tried to connect to each peer twice and failed) [0]
```

**Comment by haf** [comment]
How would I go about saving the cluster?

**Comment by haf** [comment]
Secondary servers not responding:

```
➜  deis-vagrant git:(docs/cadvisor) ✗ export DEISCTL_TUNNEL=172.17.8.101
➜  deis-vagrant git:(docs/cadvisor) ✗ deisctl config platform set domain=local3.deisapp.com
```

**Comment by bacongobbler** [comment]
If you want to save the cluster's data, use http://docs.deis.io/en/latest/managing_deis/backing_up_data/ to back up your data.

**Comment by haf** [comment]
It's just for test, I'd have that in 'real' production; but given that two nodes are alive, shouldn't it be possible to save the situation?

**Comment by haf** [comment]
Trying to reboot deis-02, but then the VM crashes hard (aborted):

```
reboot
Failed to execute operation: Interactive authentication required.
Must be root.
core@deis-02 ~ $ sudo !!
sudo reboot
Connection to 127.0.0.1 closed by remote host.
Connection to 127.0.0.1 closed.
➜  deis-vagrant git:(docs/cadvisor) ✗ vagrant ssh deis-02
VM must be running to open SSH connection. Run `vagrant up`
to start the virtual machine.
➜  deis-vagrant git:(docs/cadvisor) ✗ vagrant status deis-02
Current machine states:

deis-02                   aborted (virtualbox)

The VM is in an aborted state. This means that it was abruptly
stopped without properly closing the session. Run `vagrant up`
to resume this virtual machine. If any problems persist, you may
have to destroy and restart the virtual machine.
```

**Comment by bacongobbler** [comment]
yes it's possible. WRT the fleet errors I'd look at the fleet documentation to see how you can recover from a node failure, but as long as you still hold quorum you should be fine.

As for the hardware requirements to run Deis locally, there's http://docs.deis.io/en/latest/installing_deis/system-requirements/

**Comment by haf** [comment]
In this test I've killed off the quorum, after the restart I'm getting interesting errors "Raft Internal Error"!

```
Mar 08 16:46:43 deis-02 etcd[568]: [etcd] Mar  8 16:46:43.459 INFO      | 28ed86924a1c476b9f88ec56074a5a1c attempted to join via 172.17.8.102:7001 failed: fail on join request: Raft Internal Error ()
Mar 08 16:46:43 deis-02 etcd[568]: [etcd] Mar  8 16:46:43.459 INFO      | 28ed86924a1c476b9f88ec56074a5a1c attempted to join via 172.17.8.100:7001 failed: fail checking join version: Client Internal Error (Get http://172.17.8.100:7001/version: dial tcp 172.17.8.100:7001: connection refused)
```

**Comment by bacongobbler** [comment]
If you've killed quorum then you've lost quorum, which etcd has no form of recovery from that. For reference, coreos/etcd#1994

**Comment by haf** [comment]
In fact, there seems to be an internal error condition in the Raft implementation that Etcd uses, because now the term is UInt64.MaxValue:

```
Mar 08 16:50:11 deis-03 etcd[943]: [etcd] Mar  8 16:50:11.184 INFO      | 169914f25a9b47f38a8f4da8eee9a714: term #18446744073709551615 started.
Mar 08 16:50:11 deis-03 etcd[943]: [etcd] Mar  8 16:50:11.184 INFO      | 169914f25a9b47f38a8f4da8eee9a714: state changed from 'follower' to 'stopped'.
Mar 08 16:50:11 deis-03 etcd[943]: [etcd] Mar  8 16:50:11.184 INFO      | 169914f25a9b47f38a8f4da8eee9a714: state changed from 'stopped' to 'stopped'.
```

No wonder it can't make forward progress. The paper recommended 300 ms between each election; how come etcd doesn't do that?

**Comment by haf** [comment]
I hope you're alright with me documenting a how I go through disaster recovery? Better now than in production, right?

So, following [the docs](https://github.com/coreos/etcd/blob/master/Documentation/admin_guide.md#disaster-recovery) for Disaster Recovery for etcd, I'm supposed to do a backup. That doesn't work:

```
$ sudo  etcdctl backup --data-dir /var/lib/etcd --backup-dir /tmp/etcd_backup
No help topic for 'backup'
```

**Comment by haf** [comment]
Well, that was that. Disaster recovery works by closing the issue :)

**Comment by bacongobbler** [comment]
You shouldn't need to back up etcd, but if you want some keys that you set via `deisctl`, you can use https://github.com/AaronO/etcd-dump.

ref: https://github.com/coreos/etcd/blob/v0.4.6/Documentation/upgrade.md#upgrading-an-existing-cluster

**Comment by haf** [comment]
@bacongobbler thank you for your very quick replies! Is there a command I can run to upgrade the binary?

**Comment by bacongobbler** [comment]
CoreOS is a managed operating system so there's no command to upgrade etcd directly, sadly

**Comment by ianblenke** [comment]
Even the latest CoreOS alpha channel version 612 uses etcd 0.4.7

For Deis to adopt etcd 2.0 while CoreOS itself is using 0.4.7 would require running a parallel version of fleet and etcd, which would greatly complicate things for the sole purpose of running a version of etcd that CoreOS itself has yet not adopted.

Until CoreOS addresses their etcd 2.0 private API reservations, it would be foolhardy for Deis to attempt a workaround for this.

**Comment by bacongobbler** [comment]
Bringing back the discussion to the OP...

Does the docs at http://docs.deis.io/en/latest/managing_deis/backing_up_data/ provide enough information to help get you started with backing up your data, or is there something missing? I'm thinking that we should close this as the original issues reported are CoreOS/etcd/fleet failures, not necessarily Deis. I'm not quite sure what's the actionable item here on our end.

**Comment by haf** [comment]
The Etcd guys said that the Deis guys should try out v2.0 - what's your take on this?

**Comment by carmstrong** [comment]
> The Etcd guys said that the Deis guys should try out v2.0 - what's your take on this?

Unfortunately that's what we've been hearing for months, but until etcd v2.0 makes its way into the CoreOS stable release channel, it's not something we're comfortable shipping to our users.

I have tested etcd 2.0 briefly as a drop-in replacement, and have confirmed that it works out-of-the-box, but haven't done any reliability testing.

**Comment by haf** [comment]
I agree with you, it's better to use 'stable', because that's hopefully what it means. I think you're right in trying out the new version but not pushing it to everyone.

I guess I have to be very vigilant with the backups from ceph, and I think I could have solved the issue by re-generating the discovery key. It's a bit sad that it takes many months for 2.0 to reach users though.

**Comment by azurewraith** [comment]
I tried pretty painfully to bring etcd back from the dead to no avail, even with the current Deis docs.  The discovery URL gets embedded into the system unit and doesn't get overwritten anymore by cloud-config.  Oddly, this used to work.  I think there needs to be some sort of documentation around recovering etcd if you've lost quorum.  There should be a way to just tell your cluster to fetch a new discovery URL and regain quorum if you tell every host to abandon ship.

...and @bacongobbler, I only saw a blurb about restoring etcd keys and not capturing them in that link above.  I thought it was odd yesterday.
